### PR TITLE
GH-1108 AmCharts Combo

### PR DESCRIPTION
**Pull request by mlzummo**
Adds Amcharts Combo Chart

Fixes GH-1108

Signed-off-by: Mathew Zummo <mzummo@gmail.com>

### CODE REVIEW

**Comment by mlzummo** [comment]
@GordonSmith @dtsnell4 @jbrundage - Please Review

File: src/amchart/Combo.js [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+    Combo.prototype.enter = function(domNode, element) {
+        CommonSerial.prototype.enter.apply(this, arguments);
+    };
+
+    Combo.prototype.updateChartOptions = function() {
+        CommonSerial.prototype.updateChartOptions.apply(this, arguments);
+
+        //this._chart.depth3D = this.Depth3D();
+        //this._chart.angle = this.Angle3D();
```
Comment (dtsnell4) [inline src/amchart/Combo.js]: Is this being saved for later or remove?

**Comment by dtsnell4** [comment]
@mlzummo Does this mean Amcharts got back to you?  Can you provide a link to a demo?

**Comment by mlzummo** [comment]
http://rawgit.com/mlzummo/Visualization/AMCHART_COMBO_2/demos/dermatology.html?src/amchart/Combo

**Comment by GordonSmith** [comment]
@mlzummo, @jbrundage can you compare the test factory instantiation code for your two implementations and see how similar the API calls are (I would expect them to be the same)?

**Comment by mlzummo** [comment]
@GordonSmith the api calls are the same for multi Y we use y2 and the combo charts are just combo charts and i think they have the y2 option as well

**Comment by GordonSmith** [comment]
Can you add URLs to both so I can see?

**Comment by mlzummo** [comment]
@GordonSmith 
http://rawgit.com/mlzummo/Visualization/AMCHART_COMBO_2/demos/dermatology.html?src/amchart/Combo

**Comment by GordonSmith** [comment]
@mlzummo I am looking for something similar to:
https://github.com/hpcc-systems/Visualization/blob/master/test/amchartFactory.js#L48-L65

**Comment by mlzummo** [comment]
@GordonSmith can we take a look at this again?

File: src/amchart/Combo.js [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+                gObj.bullet = this.bulletType();
+                gObj.bulletSize = this.bulletSize();
+            }
+
+            gObj.valueField = this.columns()[i + 1];
+            
+            gObj.type = gObj.type === "area" ? "line" : gObj.type; // an area chart is a line chart with fillOpacity set
+            return gObj;
+        }
+
```
Comment (mlzummo) [inline src/amchart/Combo.js]: extra space?

**Comment by mlzummo** [comment]
add this to multichart so we can show we can switch between em

**Comment by GordonSmith** [comment]
@mlzummo @jbrundage please confirm if the two combo charts are interchangeable.

**Comment by mlzummo** [comment]
test with changes to multichart branch for easy switching

**Comment by mlzummo** [comment]
the c3 and amchart are using the same data ... google we will need to make a small modification based off the factories

**Comment by mlzummo** [comment]
made PR for google change. (for some reason i cant get the gantt some of the new charts working with multichartsurface)

**Comment by GordonSmith** [comment]
@mlzummo Did you want this pulled into master or candidate-1.10.0?

**Comment by mlzummo** [comment]
closing in favor of new PR

**Pull request closed without merging**